src/templates.ts

```diff
@@ -79,16 +89,47 @@ export const templates = {
   
       protected function register_controls()
       {
+        ${CONFIG.wordpressPluginSettings.widgetSettings?.map((section) => {
+          return `
+          $this->start_controls_section(
+            '${section.id}',
+            [
+              ${Object.entries(section.options ?? {}).map(([key, value]) => {
+                return key === 'tab' || typeof value !== 'string' ? `'${key}' => ${value}` : `'${key}' => '${value}'`
+              }).join(',\n')}
+            ]
+          );
+
+          ${section.controls.map((control) => {
+            return `
+            $this->${control.responsive ? 'add_responsive_control' : 'add_control'}(
+              '${control.name}',
+              [
+                ${Object.entries(control.options ?? {}).map(([key, value]) => {
+                  return key === 'type' || typeof value !== 'string' ? `'${key}' => ${value}` : `'${key}' => '${value}'`
+                }).join(',\n')}
+              ]
+            );
+            `
+          }).join('\n')}
+
+          $this->end_controls_section();
+          `
+        }).join('\n') ?? ''}
```

Comment:
use if (hassetting-condition...) in the beginning to increase readability via early exit

dont have the user go to the very end in order to see the other path